### PR TITLE
Compatibility with Rider 2022.3 snapshot

### PR DESCRIPTION
**Pull request by maartenba**
Seeing a compile error, any help would be appreciated.
I assume it's the aspectj weaving failing because of a Gradle update + update to Java 17.

```
[error] target JDK should be comprised in between '1.1' and '1.4': 17
```

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle [hunk 1]
```diff
@@ -11,11 +11,36 @@ sourceSets {
...
+        exclude group: "com.microsoft.azure", module: "azure-client-runtime"
+        exclude group: "javax.xml.bind", module: "jaxb-api"
+    }
+    implementation 'com.microsoft.azuretools:hdinsight-node-common:3.50.0', {
+        exclude group: "com.microsoft.azure", module: "azure-client-authentication"
+        exclude group: "com.microsoft.azure", module: "azure-client-runtime"
+        exclude group: "javax.xml.bind", module: "jaxb-api"
+    }
+    implementation group: 'com.microsoft.azure', name: 'azure-auth-helper', version: '0.6.0'
+    // ---
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle]: This shouldn't be here I think, but somehow seems to be the only way to make references work.

File: PluginsAndFeatures/azure-toolkit-for-intellij/.idea/gradle.xml [hunk 2]
```diff
@@ -8,7 +8,6 @@
         <option name="testRunner" value="GRADLE" />
         <option name="distributionType" value="WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="corretto-11" />
```
Comment (ForNeVeR) [inline PluginsAndFeatures/azure-toolkit-for-intellij/.idea/gradle.xml:11]: _Maybe_ it'll be a good idea to put `gradleJvm` = `corretto-17` here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsole.kt [hunk 3]
```diff
@@ -37,7 +36,8 @@ class SparkScalaLivyConsole(module: Module) : ScalaLanguageConsole(module), Spar
         super.attachToProcess(processHandler)
 
         // Remove self from ScalaConsoleInfo
-        ScalaConsoleInfo.disposeConsole(this)
+        // TODO[JetBrains]: Figure out how to replace this.
+        // ScalaConsoleInfo.disposeConsole(this)
```
Comment (ForNeVeR) [inline PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsole.kt:40]: This has temporarily been commented-out. Discussed with the Scala team, and they'll restore this method in a new SDK build. I have a note to uncomment it later.

**Review by ForNeVeR — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle (hunk 1 above)
Comment (ForNeVeR) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/build.gradle]: I have been able to move all of dependencies inside the `mainProjects` block, so it is somewhat resolved.

**Pull request merged**